### DISCUSSION TITLE
Help: Object converted to String in route params in latest Quasar Version

### DISCUSSION THREAD

**Discussion by Lalrinfela** [quasarframework/quasar#10030] [General - Components / Directives / etc]
In the latest quasar release  `"@quasar/app": "^3.0.0"` , the router push can no longer accepts object as its parameter. It always converts to string type which was not the case before. Is there any way to pass object in the router params. I have gone though the vue router documentation and there is nothing about this.

For example: Index.vue

```
<template>
    <q-btn @click="$router.push({name: 'about'})">About</q-btn>
    <q-btn @click="$router.push({name: 'contact-us', params: {owner: model}})">Contact Us</q-btn>
  </q-page>
</template>
<script>
export default {
  name: 'PageIndex',
  data() {
    return {
      model: {
        name: 'John  Doe',
        code: '007'
      }
    }
  }
}
</script>
```

ContactUs.vue
```
<template>
  <p>Contact us Page</p>
  <q-btn @click="$router.push({name: 'home'})" label="Back"></q-btn>
</template>

<script>
export default {
  name: "ContactUs",
  props: {
    owner: {
      type: Object
    }
  },
  created() {
    console.log(this.owner) // This outputs [object object]
  }
}
</script>
```

routes.js
```
{
        name: 'contact-us',
        path: 'contact-us',
        component: () => import('pages/ContactUs.vue'),
        meta: {isProtected: false},
        props: true
      }
```

**Comment by ontwikkelfabriek** [comment]
Should be possible according to the Vue-router v4  doc:
https://next.router.vuejs.org/api/#routelocationraw

Does this work?

```javascript
<template>
  <p>Contact us Page</p>
  <q-btn @click="onClick" label="Back"></q-btn>
</template>

<script>
export default {
  name: "ContactUs",
  props: {
    owner: {
      type: Object
    }
  },
  methods: {
     onClick() {
        this.$router.push({name: 'home'})
     }
  },
  created() {
     console.log(this.owner) // This outputs [object object]
  }
}
</script>
```

**Reply by Lalrinfela** [reply]
Specifying the object property in the  push params parameters work. But sometimes the object the needs to be passed through have many properties and it's not practical to specify all properties in the params, rather just get from the data options api. This is what confuses me. Manually declaring all object properties in the params work but referring data object do not.

This works:
`this.$router.push({name: 'home', params: {rollNo: user.rollNo, class: user.class}})`


This Does not:
```
this.$router.push({name: 'home', {params: {user}})
<script>
export default {
  name: "ContactUs",
  data() {
   return {
     user: {rollNo: 'A123', class: 'Grade 9'}
   }
  }
}
</script>
```

**Reply by metalsadman** [reply]
Report it to vue router repo, if you think that's a bug.

**Reply by Lalrinfela** [reply]
I tried, but vue router issue helper is only limited to version 3.5.2 at the moment. The problem i am facing now is with vue router 4 which quasar use by default.

**Reply by metalsadman** [reply]
> I tried, but vue router issue helper is only limited to version 3.5.2 at the moment. The problem i am facing now is with vue router 4 which quasar use by default.

Googled and got https://github.com/vuejs/vue-router-next

**Reply by Lalrinfela** [reply]
Thanks @metalsadman I have submitted an issue here: https://github.com/vuejs/vue-router-next/issues/1038

**Reply by Lalrinfela** [reply]
@metalsadman It seems passing object in route params is no longer possible with router 4. We have to declare all params in the route path or pass just array or string.


For anyone who come access the same problem here is what i do:
1. I convert the object to string using JSON.stringify(object) and pass the json in the params.
2. Then in the destination page, convert the string back to object using JSON.parse(string)

Don't know why the object passing feature is removed in Vue router 4. It was very useful at least in all my projects and use cases. Now i have to add a little more steps.

**Reply by metalsadman** [reply]
> Don't know why the object passing feature is removed in Vue router 4. It was very useful at least in all my projects and use cases. Now i have to add a little more steps.

yep, i didn't find any mention of this in their docs too (or maybe i just missed it).

**Reply by ontwikkelfabriek** [reply]
@metalsadman 

I don't think it's removed. This is the **V4** doc of vue-router. It does mention:
_A raw location can either be a **string** like /users/posva#bio or an **object**:_

https://next.router.vuejs.org/api/#routelocationraw
